timestamp-impl/src/test/java/com/palantir/timestamp/PersistentTimestampServiceTest.java

```diff
@@ -71,6 +73,52 @@ public void testFastForward() {
     }
 
     @Test
+    public void incrementUpperLimitIfOneMinuteElapsedSinceLastUpdate() throws InterruptedException {
+        Clock clock = mock(Clock.class);
+        when(clock.getTimeMillis()).thenReturn(0L, TWO_MINUTES_IN_MILLIS, 2 * TWO_MINUTES_IN_MILLIS, 3 * TWO_MINUTES_IN_MILLIS);
+        TimestampBoundStore timestampBoundStore = initialTimestampBoundStore();
+        PersistentTimestampService persistentTimestampService = PersistentTimestampService.create(timestampBoundStore, clock);
+
+        persistentTimestampService.getFreshTimestamp();
+        Thread.sleep(10);
+        persistentTimestampService.getFreshTimestamp();
+        Thread.sleep(10);
+        verify(timestampBoundStore, times(2)).storeUpperLimit(anyLong());
+    }
+
+    @Test
+    public void incrementUpperLimitOnFirstFreshTimestampRequest() {
+        TimestampBoundStore timestampBoundStore = initialTimestampBoundStore();
+        PersistentTimestampService persistentTimestampService = PersistentTimestampService.create(timestampBoundStore);
+
+        persistentTimestampService.getFreshTimestamp();
+
+        verify(timestampBoundStore).storeUpperLimit(PersistentTimestampService.ALLOCATION_BUFFER_SIZE);
+    }
+
+    @Test
+    public void multipleFreshTimestampRequestsShouldIncreaseUpperLimitOnlyOnce() {
+        TimestampBoundStore timestampBoundStore = initialTimestampBoundStore();
+        PersistentTimestampService persistentTimestampService = PersistentTimestampService.create(timestampBoundStore);
+
+        getFreshTimestampsInParallel(persistentTimestampService, 20);
+
+        verify(timestampBoundStore, times(1)).storeUpperLimit(PersistentTimestampService.ALLOCATION_BUFFER_SIZE);
+    }
+
+    @Test
```

Comment:
I actually want the allocateMoreTimestamps call (on the timeboundStore) to happen on first request, and not on initialization (see PR description for motivation). 

Would it be clearer if I rename this test to something like `notAllocateTimestampsOnInitialization()`?